lisa/sut_orchestrator/baremetal/build.py

```diff
@@ -79,23 +105,54 @@ def copy(self, sources_path: List[Path], files_map: List[FileSchema]) -> None:
                         )
                 self._log.debug(f"copy file {file} to {share_name}\\{file_name}")
 
-    def _find_matched_files(
-        self, sources_path: List[Path], files_map: List[FileSchema]
-    ) -> Dict[str, FileSchema]:
-        all_files = []
-        match_files: Dict[str, FileSchema] = {}
-        for source_path in sources_path:
-            for root, _, files in os.walk(source_path):
-                for file in files:
-                    all_files.append(os.path.join(root, file))
-
-            for file_map in files_map:
-                file_path = rf"{source_path}\{file_map.source}".replace("\\", "\\\\")
-                pattern = re.compile(
-                    file_path,
-                    re.I | re.M,
+
+class TftpBuild(Build):
+    def __init__(self, runbook: TftpBuildSchema) -> None:
+        super().__init__(runbook)
+        self.pxe_runbook: TftpBuildSchema = self.runbook
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "tftp"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return TftpBuildSchema
+
+    def copy(self, sources_path: List[Path], files_map: List[FileSchema]) -> None:
+        assert self.pxe_runbook.connection, "The build server is not specified"
+
+        try:
+            build_server = quick_connect(
+                self.pxe_runbook.connection,
+                logger_name="build_server",
+            )
+        except Exception as ex:
+            self._log.debug(
+                f"failed to connect {self.pxe_runbook.connection}",
+                exc_info=ex,
+            )
+
+        ls = build_server.tools[Ls]
+        rc = build_server.tools[RemoteCopy]
+        for file, file_map in _find_matched_files(
+            sources_path,
+            files_map,
+        ).items():
+            if file_map.destination:
+                if ls.is_file(PurePath(file_map.destination)):
+                    file_destination = os.path.dirname(PurePath(file_map.destination))
+                else:
+                    file_destination = file_map.destination
+            else:
+                file_destination = os.path.dirname(PurePath(file))
```

Comment:
1. You don't need to convert between PurePath and string forth and back. Just keep a PurePath, and pass it into the `copy_to_remote`.
2. For remote path, it needs to call `Node.get_pure_path` to get pure path.

Reply:
The change is done, thanks